src/ui/dialogs/spring_pane.rb

```diff
@@ -39,19 +112,54 @@ def open_dialog
     @dialog.set_html(t.result(binding))
     @dialog.set_position(500, 500)
     @dialog.show
-    register_insights_callbacks
+    register_callbacks
+  end
+
+  # compilation / simulation logic:
+
+  def try_compile
+    @simulation_runner ||= SimulationRunner.instance
+    @simulation_data ||= simulate
```

Comment:
This is double, cause simulate already writes into @simulation_data

Reply:
Fixed in 6d9d15a